src/Symfony/Component/HttpFoundation/ProblemDetailsJsonResponse.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+use Symfony\Component\HttpFoundation\Exception\ProblemDetailsJsonResponseException;
+
+/**
+ * Represents a JSON response with a Problem Details object.
+ *
+ * @author Abdellah Ramadan <ramadanabdel24@gmail.com>
+ */
+class ProblemDetailsJsonResponse extends Response
+{
+    public function __construct(
+        private ?int $status = null,
+        private ?string $title = null,
+        private readonly ?string $type = null,
+        private readonly ?string $detail = null,
+        private readonly ?string $instance = null,
+        private readonly ?array $extensions = [],
+    ) {
+        parent::__construct();
+
+        $this->status = $this->status ?? 500;
+        $this->statusCode = $this->status;
+
+        if ($this->status < 400 || $this->status > 599) {
+            throw new ProblemDetailsJsonResponseException(\sprintf('The status code "%s" is not a valid HTTP Status Code error.', $this->statusCode));
+        }
+
+        if ($this->title && null === $this->type) {
+            $this->title = Response::$statusTexts[$this->status];
```

Comment:
it looks strange to me to override the title if it's already set before

Reply:
If the problem type is not specified, the status code text is used as the title

> When "about:blank" is used, the title SHOULD be the same as the recommended HTTP status phrase for that code (e.g., "Not Found" for 404, and so on), although it MAY be localized to suit client preferences (expressed with the Accept-Language request header) 

[Per the RFC](https://datatracker.ietf.org/doc/html/rfc9457#section-4.2.1-4)